examples/text-generation/run_generation.py

```diff
@@ -59,9 +63,61 @@ def check_optimum_habana_min_version(*a, **b):
 logger = logging.getLogger(__name__)
 
 
-def main():
+def setup_env(args):
+    if args.global_rank == 0:
+        os.environ.setdefault("GRAPH_VISUALIZATION", "true")
+        shutil.rmtree(".graph_dumps", ignore_errors=True)
+
+    if args.world_size > 0:
+        os.environ.setdefault("PT_HPU_LAZY_ACC_PAR_MODE", "0")
+        os.environ.setdefault("PT_HPU_ENABLE_LAZY_COLLECTIVES", "true")
+
+    # Tweak generation so that it runs faster on Gaudi
+    from optimum.habana.transformers.modeling_utils import adapt_transformers_to_gaudi
+
+    adapt_transformers_to_gaudi()
+
+
+def setup_device(args):
+    return torch.device(args.device)
+
+
+def initialize_model(args):
+    init_start = time.perf_counter()
+    setup_distributed(args)
+    override_prints(args.global_rank == 0 or args.verbose_workers, logger)
+    setup_env(args)
+    setup_device(args)
+    set_seed(args.seed)
+    get_repo_root(args.model_name_or_path, local_rank=args.local_rank, token=args.token)
+    use_deepspeed = args.world_size > 0
+    if use_deepspeed or args.bf16:
+        model_dtype = torch.bfloat16
+    else:
+        model_dtype = torch.float
+        args.attn_softmax_bf16 = False
+
+    model_kwargs = {
+        "revision": args.model_revision,
+        "token": args.token,
+    }
+    model = (
+        setup_model(args, model_dtype, model_kwargs)
+        if not use_deepspeed
+        else setup_distributed_model(args, model_dtype, model_kwargs)
+    )
+    tokenizer, model = setup_tokenizer(args, model)
+    generation_config = setup_generation_config(args, model, tokenizer)
+    init_end = time.perf_counter()
+    logger.info(f"Args: {args}")
+    logger.info(f"device: {args.device}, n_hpu: {args.world_size}, bf16: {model_dtype == torch.bfloat16}")
+    logger.info(f"Model initialization took {(init_end - init_start):.3f}s")
+    return model, tokenizer, generation_config
+
+
+def setup_parser(parser):
```

Comment:
This one we keep it here as it will be easier to have all arguments visible in the same file